app/frontend/wallet/account-manager.ts

```diff
@@ -46,29 +48,68 @@ const AccountManager = ({
   }
 
   async function discoverAccounts() {
+    if (
+      maxAccountIndex === accounts.length - 1 ||
```

Comment:
nitpick, just in case, I'd rather check for `<=` rather than strict equality
```suggestion
       accounts.length - 1 >= maxAccountIndex ||
```

Reply:
Refactored.